### DISCUSSION TITLE
imapsync manual Alpine installation

### DISCUSSION THREAD

**Discussion by mpsarakis** [imapsync/imapsync#470] [Show and tell]
Hello,

In case somebody is interested, these are the Alpine packages required if anybody wants to install the latest imapsync release on their light-weight system:
```
echo "**** Install IMAPSYNC required packages ****"
apk --no-cache --no-progress add \
    make                         \
    gcc                          \
    git                          \
    db-dev                       \
    openssl-dev                  \
    perl-app-cpanminus           \
    lsb-release-minimal          \
    procps

apk --no-cache --no-progress add \
    perl-ntlm                    \
    perl-class-load              \
    perl-crypt-ssleay            \
    perl-data-uniqid             \
    perl-digest-hmac             \
    perl-dist-checkconflicts     \
    perl-encode-imaputf7         \
    perl-file-copy-recursive     \
    perl-file-tail               \
    perl-io-socket-inet6         \
    perl-io-socket-ssl           \
    perl-io-tee                  \
    perl-mail-imapclient         \
    perl-module-scandeps         \
    perl-net-ssleay              \
    perl-par-packer              \
    perl-regexp-common           \
    perl-sys-meminfo             \
    perl-term-readkey            \
    perl-test-fatal              \
    perl-test-mock-guard         \
    perl-test-mockobject         \
    perl-test-pod                \
    perl-test-requires           \
    perl-test-simple             \
    perl-unicode-string          \
    perl-test-nowarnings         \
    perl-test-deep               \
    perl-test-warn               \
    perl-readonly                \
    perl-crypt-openssl-rsa       \
    perl-package-stash-xs        \
    perl-proc-processtable       \
    perl-cgi                     \
    perl-json                    \
    perl-json-webtoken           \
    perl-net-server
```

These are packages to be able to use OAUTH2 in case you need it:
```
echo "**** Install IMAPSYNC OAUTH2 required packages ****"
apk --no-cache --no-progress add \
    unzip                        \
    perl-net-dns                 \
    perl-email-address           \
    perl-http-daemon
```

It would have certainly been possible to use the older existing Alpine package for Imapsync + replace the actual script by the latest one but I wanted to keep it fully manual since package was outdated anyway... maybe not the ideal way but working fine here and avoid having to compile any perl module.

There are possibly a few packages for my personal installation needs in the lists but nothing really bothersome.

Thought it may still help somebody...

**Comment by gilleslamiral** [comment]
Thanks!

Just added
https://imapsync.lamiral.info/INSTALL.d/INSTALL.AlpineLinux.txt

**Comment by mpsarakis** [comment]
Hello Gilles,

It seems you have changed the oauth2 programs recently in:
https://imapsync.lamiral.info/oauth2/oauth2_imap.zip

And, using Alpine, the execution of ./source/oauth2_imap/oauth2_imap fails with this error:
Can't locate HTTP/Daemon/SSL.pm in @INC (you may need to install the HTTP::Daemon::SSL module) (@INC entries checked: /usr/local/lib/perl5/site_perl /usr/local/share/perl5/site_perl /usr/lib/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/core_perl /usr/share/perl5/core_perl) at oauth2/source/oauth2_imap/oauth2_imap line 17.
BEGIN failed--compilation aborted at oauth2/source/oauth2_imap/oauth2_imap line 17.

Comparing it to the older version, this line was commented out in the old version as follow:
#use HTTP::Daemon::SSL ; # does not work

By commenting it in the new version, it seems to function correctly...

Can I leave it like that? Is that something that could be left commented for everybody (so Alpine would then work natively) or does that remove needed features somewhere ?

Thanks for your help.

**Comment by gilleslamiral** [comment]
> By commenting it in the new version, it seems to function correctly

It compiles but it will fail functionally when a localhost https server is needed by the OAUTH2 authentication.
It would be better to add the Alpine package giving the Perl  module HTTP::Daemon::SSL

**Reply by mpsarakis** [reply]
OK, I see.

Actually, I had tried to find the missing module but it seems it does not exist as a package with Alpine:
https://pkgs.alpinelinux.org/contents?file=SSL.pm&path=*Daemon*&name=&repo=main&arch=x86

Is there an alternative package that may be used in your program (if it is easy) that would be more available ? I am not a perl dev so I have no idea if HTTP::Daemon::SSL is a mainstream package or something a bit more obscure ?

Or do you have any idea where I could find the missing package for Alpine ?

Maybe another (not ideal) option, for the time being, would be to use a previous release of your oauth module which did not require this particular package ? Is it stored somewhere in GitHub ?

Thanks!

**Comment by mpsarakis** [comment]
Sorry but I am actually not really skilled in Perl.

I had the approach to use packages only so to limit installation time (avoiding compilation) for Alpine.

I have tried to use cpanm to specifically install/compile the missing module:
`cpanm --force HTTP::Daemon::SSL
`

Now it seems to work with your new oauth2 version.

Would that be the right approach ?

If yes, you could maybe update the Alpine doc you had created to reflect this ?

**Reply by gilleslamiral** [reply]
> I have tried to use cpanm to specifically install/compile the missing module: `cpanm --force HTTP::Daemon::SSL `
> Would that be the right approach ?

Yes

> If yes, you could maybe update the Alpine doc you had created to reflect this ?

Okay, done:
https://imapsync.lamiral.info/INSTALL.d/INSTALL.AlpineLinux.txt
